pkg/registry/core/namespace/strategy.go

```diff
@@ -47,6 +48,22 @@ func (namespaceStrategy) NamespaceScoped() bool {
 	return false
 }
 
+// ResetFieldsFor returns a set of fields for the provided version that get reset before persisting the object.
+// If no fieldset is defined for a version, nil is returned.
+func (namespaceStrategy) ResetFieldsFor(version string) *fieldpath.Set {
+	set, ok := resetFieldsByVersion[version]
+	if !ok {
+		return nil
+	}
+	return set
+}
+
+var resetFieldsByVersion = map[string]*fieldpath.Set{
+	"v1": fieldpath.NewSet(
```

Comment:
For each of these, you have to carefully look at the `PrepareForCreate` and `PrepareForUpdate` methods. For example, I think this is missing `spec.Finalizers`.

Reply:
I was not sure about Finalizers.
They don't get wiped, but updated or appended with the KubeFinalizer if it's not there. Which means, the user can own the finalizers, as they don't get wiped on create.

But they do on update, so there's a difference between Create and Update.
We probably can add them to the list, because after being created they can not be owned through updates.

Also i didn't see there is a finalize strategy.... Should handle that as well.
Tbh when starting this, i expected it to be way more mechanical and generic to do :-/

Reply:
Note that metadata finalizers and namespace finalizers are sadly distinct concepts. I'm not sure why either one would get wiped. It should be fine for users or controllers to own finalizers.